### PR TITLE
Rewritten, fixed a few glitches

### PR DESCRIPTION
**Pull request by yunruse**
I made the algorithm precise to the second, and made the command-line output input zenith and output in a slightly more extractable format. I also rewritten the main calculation to become a lot smaller.

While it's now a tiny bit slower because it must re-do a few calculations, considering the only difference in the sunrise and sunset is in two lines, I'd say this algorithm is a wee bit easier on the eye.

### CODE REVIEW

File: sunrise_sunset.py [hunk 1]
```diff
@@ -1,150 +1,111 @@
...
-
-    print("Latitude: ", latitude)
-    print("Longitude: ", longitude)
-    print("Date: ", date)
-    print("Offset: ", localOffset)
-    print("Sunrise", rise_time)
-    print("Sunset", set_time)
+    args = parser.parse_args()
+    lat, long = args.latitude, args.longitude
+    date = datetime.striptime(args.date, '%Y-%m-%d') if args.date else datetime.now()
```
Comment (CarvellScott) [inline sunrise_sunset.py]: This error is encountered when you use the "-d" argument:
`AttributeError: type object 'datetime.datetime' has no attribute 'striptime'`

**Review by CarvellScott — CHANGES_REQUESTED** [review]
I like the intention of simplifying the formula, it's a great idea! I didn't even think about `datetime.timetuple().tm_yday`. I also like that you eliminated a non-standard-library dependency. However, some inaccuracy has been introduced.

At the time of writing (2018-11-03), Google predicts sunset for my location (San Antonio, 29.4241° N, 98.4936° W) at about 18:46. With these changes, the script says sunset happens at 00:56:01. Even if that is UTC instead of local time, the minutes are quite a bit off. Unless of course Google's wrong here. If that's the case, might I ask where you got the value of "CIVIL_ZENITH" from?

Also, there's an AttributeError encountered when you use the -d parameter.

**Comment by yunruse** [comment]
Hi,

I believe you might have my commits confused for someone else's – if you check the diff, I didn't touch the ZENITH value, fix any imports, or use `.tm_yday`. The attribute error was easily fixed — a simple typo.

However, as far as I can tell after a few careful reads, there is no mathematical difference between the formula I used and the previous versions – only a change in precision. Debugging and printing values seems to show consistency, so as far as I can tell any errors are due to previous commits, apologies.

**Comment by CarvellScott** [comment]
Odd. I could've swore I saw that in the diff somewhere. I must have messed up the comparison somehow. Sorry.

**Pull request closed without merging**